src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/LanguageServerProjectSystem.cs

```diff
@@ -205,6 +206,15 @@ private async Task<(LSP.MessageType? failureType, BuildHostProcessKind? preferre
             if (await buildHost.IsProjectFileSupportedAsync(projectPath, cancellationToken))
             {
                 var loadedFile = await buildHost.LoadProjectFileAsync(projectPath, cancellationToken);
+                var diagnosticLogItems = await loadedFile.GetDiagnosticLogItemsAsync(cancellationToken);
+                LogDiagnostics(projectPath, diagnosticLogItems);
```

Comment:
One subtle bit: unfortunately how the existing code works is all the diagnostics created for loading get dumped into the same collection, both for evaluation and execution passes. So you probably only want to call LogDiagnostics() if you're in your if check and bailing; otherwise you can have the later LogDiagnostics() call.

We should also make LogDiagnostics() just return the most severe type of LSP.MessageType so we can inline that.